tests/src/main/scala/tests.scala

```diff
@@ -176,6 +176,14 @@ object Character {
   type Id = Long with Tag
 }
 
+case class AnotherCharacter(id: AnotherCharacter.Id)
+object AnotherCharacter {
+  trait Tag extends Any
+  type Id = Long with Tag
+
+  implicit val idShow: Show[String, Id] = _.toString
```

Comment:
Hmm is this instance actually needed?

Reply:
Hi @joroKr21 yeah it's needed otherwise we will hit this [case]( https://github.com/propensive/magnolia/blob/fa8b0d420f425fc97a37ee67ae8d2fc45e0d92c6/tests/src/main/scala/tests.scala#L653)